crates/matrix-sdk-crypto/src/identities/device.rs

```diff
@@ -69,6 +71,32 @@ pub enum MaybeEncryptedRoomKey {
     },
 }
 
+#[derive(Clone, Copy, Default, Serialize, Deserialize)]
+#[serde(from = "u64", into = "u64")]
+struct SecondsSinceUnixEpoch(u64);
+
+impl SecondsSinceUnixEpoch {
+    fn now() -> Self {
+        SecondsSinceUnixEpoch(
+            SystemTime::now()
+                .duration_since(SystemTime::UNIX_EPOCH)
+                .map_or(0f64, |t| t.as_secs_f64())
+                .to_bits(),
+        )
+    }
+}
+
+impl From<u64> for SecondsSinceUnixEpoch {
+    fn from(value: u64) -> Self {
+        SecondsSinceUnixEpoch(cmp::max(0, value))
```

Comment:
```suggestion
        Self(value)
```

It's `u64`, it's impossible to get a negative value, hence `max(0, value)` is not necessary.

Reply:
oh it used to be a f64 :)